File: Go/Password Strength tester/Password Strength Tester.go
package main

import (
	"fmt"
	"strings"
)

func lengthTest(passwd, passed, failed string) string {
	minimumLength := 12
	lengthResult := ""
	if len(passwd) >= minimumLength {
		lengthResult = passed
		return lengthResult
	} else {
		lengthResult = failed
		return lengthResult
	}
}

func specialTest(passwd, passed, failed string) string {
	specialResult := ""
	var specialChara = "!#$%&\\'()*+,-./:;<=>?@[\\]^_`{|}~"
	if strings.ContainsAny(passwd, specialChara) {
		specialResult = passed
		return specialResult
	} else {
		specialResult = failed
		return specialResult
	}

}

func numbersTest(passwd, passed, failed string) string {
	numbersResult := ""
	var numbers = "0123456789"
	if strings.ContainsAny(passwd, numbers) {
		numbersResult = passed
		return numbersResult
	} else {
		numbersResult = failed
		return numbersResult
	}

}

func main() {
	var passed = "✓"
	var failed = "✗"

	var passToTest string
	fmt.Print("Enter a password to test: ")
	fmt.Scan(&passToTest)
	fmt.Printf("Length: %s\n", lengthTest(passToTest, passed, failed))
	fmt.Printf("Special Characters: %s\n", specialTest(passToTest, passed, failed))
	fmt.Printf("Numbers: %s\n", numbersTest(passToTest, passed, failed))
}

File: Python/Password Strength Tester.py
import string

special = string.punctuation
minimum_length = 12
numbers = string.digits
passed = "✓"
failed = "✗"


passwd_test = input("Enter a password to test its strength: ")

if len(passwd_test) >= minimum_length:
    length_test = passed
else:
    length_test = failed

if any(char in special for char in passwd_test):
    special_test = passed
else:
    special_test = failed

if any(char in numbers for char in passwd_test):
    number_test = passed
else:
    number_test = failed

print("Test Results")
print(f"Length: {length_test}")
print(f"Special Characters: {special_test}")
print(f"Numbers: {number_test}")
